DQMServices/Demo/test/test_dqmgui_upload.sh

```diff
@@ -0,0 +1,64 @@
+#!/bin/env bash
+# Simple test to use one of the files generated by TestDQMServicesDemo test
+# to upload it to the dev DQMGUI, making sure that the ROOT file format generated
+# is compatible. See cmssw issue #43590.
+# Requires curl and jq to be installed. Also requires either a Grid key and certificate,
+# or a proxy to be available.
+set -x
+set -e
+
+DEV_DQMGUI_URL="https://cmsweb.cern.ch/dqm/dev"
+# Create a unique fake "Era", so that different executions of this test
+# produce differently named files. This is required, because this test might pass
+# if it's checking a file that was successfully uploaded during a previous
+# instance of the same test.
+UNIQUE_ERA_ID=$(date -u +%Y%m%d%M%S%N)
+OLD_FILENAME=DQM_V0001_R000000001__Harvesting__DQMTests__DQMIO.root
+NEW_FILENAME=${OLD_FILENAME/DQMTests/DQMTests${UNIQUE_ERA_ID}}
+
+# Make sure the required file exists. We expect to find it in the current directory.
+if [ ! -f $OLD_FILENAME ]; then
+    echo "Unable to find required file ${OLD_FILENAME}!"
+    exit 1
+fi
+
+# Prepare curl args
+curl_args=""
+if [ -n "$X509_CERT_DIR" ]; then
+    curl_args="--capath ${X509_CERT_DIR}/ --key ${X509_USER_PROXY} --cert ${X509_USER_PROXY}"
+elif [ -n "$X509_USER_CERT" ] && [ -n "$X509_USER_CERT" ]; then
+    curl_args="--capath ${X509_CERT_DIR}/ --key ${X509_USER_KEY} --cert ${X509_USER_CERT}"
+elif [ -f $HOME/.globus/usercert.pem ] && [ -f $HOME/.globus/userkey.pem ]; then
```

Comment:
@nothingface0 , looks like testis not using `X509_USER_PROXY` . Although `X509_USER_PROXY` is set but due to `[ -n "$X509_CERT_DIR" ]` check script is not using it.  I would suggest to use `X509_CERT_DIR` to only add `--capath ${X509_CERT_DIR}` and separate condition for `X509_USER_PROXY`

Reply:
Any better like this?

Reply:
looks good now